src/Microsoft.Diagnostics.Runtime/src/DataTargets/DataTarget.cs

```diff
@@ -164,10 +168,10 @@ private ImmutableArray<ClrInfo> GetOrCreateClrVersions()
                     {
                         // Works around issue https://github.com/dotnet/coreclr/issues/20205
                         int processId = Process.GetCurrentProcess().Id;
-                        string tempDirectory = Path.Combine(Path.GetTempPath(), "clrmd" + processId);
-                        Directory.CreateDirectory(tempDirectory);
+                        _tempDirectory = Path.Combine(Path.GetTempPath(), "clrmd" + processId);
+                        Directory.CreateDirectory(_tempDirectory);
 
-                        string symlink = Path.Combine(tempDirectory, dacFileName);
+                        string symlink = Path.Combine(_tempDirectory, dacFileName);
```

Comment:
This design creates a subtle race between two ClrMD processes inspecting the same target process.

If Process A is in `CleanupTmpDirectory` in or just before `Directory.Delete` and Process B has just returned from `LinuxFunctions.symlink` then Process A may delete our symlink before we have a chance to open it, leading to an odd exception in Process B when it can't find the dac symlink we created.

Reply:
Aha? The temporary directory has the ClrMD process' ID in its name.

Reply:
Ah you are right, I didn't notice it's the current process and not the target process.  What if two different threads in the same process create a DataTarget and clean it up?  I'm not sure the behavior of `LinuxFunctions.symlink` will be if two threads race to call that at the same time, will that fail and we won't set the dacLocation?  I think that same cleanup issues happens if two threads in the same process races.